### PR TITLE
[No QA] Fix Comment on issues GH action.

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

 It's a follow up PR for https://github.com/Expensify/Expensify.cash/pull/3771#issuecomment-876769431
 
### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

**Comment by parasharrajat** [comment]
@roryabraham  this is an attempt to fix the break.
1. Currently I have added logging to see the real issue. 
2. I found one interesting fact that `isStagingDeployLocked` returns current opened issue with LockDeploycash label. But we don't want to get that. Instead, we should pick the StagingDeployIssue which was active when PR was merged. Any idea, how can I get that?

**Comment by roryabraham** [comment]
> Instead, we should pick the StagingDeployIssue which was active when PR was merged. Any idea, how can I get that?

Unfortunately, I don't think there's any really easy way to do it. Maybe something like this pseudocode:

```javascript
const mergedAt = PR.merged_at;
const allStagingDeployCashIssues = getAllStagingDeployCashIssues().sortBy('newestToOldest');
let stagingDeployCash = null;
for (let issue of allStagingDeployCashIssues) {
    if (issue.created < mergedAt) {
        // This is the first StagingDeployCash created before this PR was merged.
        // Therefore it was the one that was active when the PR was merged.
        stagingDeployCash = issue;
        break;
    }
}
```

**Comment by parasharrajat** [comment]
🚀 [Deployed](undefined/undefined/actions/runs/undefined) to staging         in version: test🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

**Comment by parasharrajat** [comment]
Update: I have fixed the core issue which broke the GH action in the first place. Now I am doing changes related to https://github.com/Expensify/Expensify.cash/pull/3937#issuecomment-876823066 comment.

To test the Changes I will use my old PR https://github.com/Expensify/Expensify.cash/pull/3771 and one more with CP label. Please ignore the Comments like https://github.com/Expensify/Expensify.cash/pull/3937#issuecomment-878460588. 

Thank you.

**Comment by roryabraham** [comment]
@parasharrajat Did you forget to push your changes here?

**Comment by parasharrajat** [comment]
Yeah, I am doing the last few changes to get the correct StagingDeployIssue and then I will push those changes all along.

**Comment by parasharrajat** [comment]
@roryabraham Updated and tested on https://github.com/Expensify/Expensify.cash/pull/3581. Seems to work fine. 
I have tried to minimize the Github API calls to a minimum best of my knowledge.

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 1]
```diff
@@ -34,15 +34,49 @@ function getDeployTableMessage(platformResult) {
...
+ */
+function fetchAllStagingDeployCash(fromTimestamp) {
+    return GithubUtils.octokit.paginate(GithubUtils.octokit.issues.listForRepo, {
+        owner: GithubUtils.GITHUB_OWNER,
+        repo: GithubUtils.EXPENSIFY_CASH_REPO,
+        state: 'all',
+        sort: 'created',
+        direction: 'desc',
+        labels: GithubUtils.STAGING_DEPLOY_CASH_LABEL,
+        per_page: 30,
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:60]: I think 30 per page is the default, right?

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 2]
```diff
@@ -122,21 +172,28 @@ function commentPR(pr) {
 }
 
 const run = function () {
-    return Promise.all([
-        getLockCashDeploysTimeline(),
-        GithubUtils.fetchAllPullRequests(prList),
-    ])
-        .then(([lockCashDeployLabelTimeSet, PRListWithDetails]) => {
-            lockCashDeployLabelTimeline = lockCashDeployLabelTimeSet;
+    return GithubUtils.fetchAllPullRequests(prList.map(pr => parseInt(pr, 10)))
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:175]:
```suggestion
    return GithubUtils.fetchAllPullRequests(_.compact(_.map(prList, pr => parseInt(pr, 10))))
```

Comment (roryabraham) [reply .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:175]: NAB

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js (hunk 1 above)
Comment (parasharrajat) [reply .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:60]: yeah

**Comment by parasharrajat** [comment]
Updated

**Review by roryabraham — APPROVED** [review]

**Review by NikkiWines — APPROVED** [review]
👍

**Pull request merged**